src/Shouldly/Internals/SourceCodeTextGetter.cs

```diff
@@ -57,15 +66,14 @@ private void ParseStackTrace(StackTrace? stackTrace)
                     TryFindGitRepoRoot(assemblyDirectory!, out sourceRoot);
```

Comment:
I think we'd get too many false positives, I just checked a handful of open source .NET projects and about half had the `.sln` file one level in (`Newtonsoft.Json`,  `Spectre.Console` etc...).

The git approach is quite crud, but might cover 90% of scenarios, for everything else we can tell users to pass a `SourceRoot`.

For deterministic builds, there an MSBuild `ItemGroup` item named `SourceRoot` which ideally we'd want to reuse, but the only way to get it would be some sort of source code generation.